### DISCUSSION TITLE
How to gracefully shutdown morbo from controller

### DISCUSSION THREAD

**Discussion by anmith** [mojolicious/mojo#1965] [Q&A] [answered]
Hi,
I created a web application for local use on my pc but I can't figure out how to create some sort of shutdown procedure in the controller so that opening an url like "/shutdown" will run a controller function which will do some cleanup etc and terminate morbo so the application is stopped.

**Comment by kraih** [comment, marked as answer]
Use `daemon` instead of morbo.

**Comment by anmith** [comment]
Thanks. 
As far as I understood, I changed the start to
`./script/my_app daemon`
and now
`exit(0);`
stops the script as expected. Or is there a better (recommended) way of shutting down?

**Reply by kraih** [reply]
For a cleaner shutdown there is also `Mojo::IOLoop->stop_gracefully`.
